src/main/java/edu/kis/powp/jobs2d/command/DriverCommand.java

```diff
@@ -1,17 +1,32 @@
 package edu.kis.powp.jobs2d.command;
 
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.canvas.Canvas;
+import edu.kis.powp.jobs2d.features.CanvasFeature;
+
+import java.util.logging.Logger;
 
 /**
  * DriverCommand interface.
  */
-public interface DriverCommand {
+public abstract class DriverCommand {
+
+    protected int posX = 0, posY = 0;
+    private final static Logger logger = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
 
     /**
      * Execute command on driver.
      *
      * @param driver driver.
      */
-    public void execute(Job2dDriver driver);
-    public void accept(CommandVisitor commandVisitor);
+    public void execute(Job2dDriver driver) {
+        Canvas currentCanvas = CanvasFeature.getCanvasManager().getCurrentCanvas();
```

Comment:
Changing DriverCommands is not an option. Especially, in the case of this task, we should focus on the work of drivers (which represent API to a device), not on higher-lever, and abstract representations of commands.

Moreover, introducing canvas abstraction to such interface results in breaking SRP and OCP rules.